### PR TITLE
Snap Package Support

### PR DESCRIPTION
**Pull request by mikeroyal**
Includes snapcraft.yaml to build snap.

Issue #, if available:
https://github.com/firecracker-microvm/firecracker/issues/723
Description of changes:
Added snapcraft.yaml to build snap.
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by raduweiss** [comment]
Thanks for getting this started. We don't have much experience with snaps yet, but we'll take a look.

For anyone else looking at this, any info on how you would use such a snap will help :)

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+base: core18
+
+architectures:
+  - build-on: amd64
+  - build-on: armhf
+  - build-on: arm64
+  
+parts:
+  firecracker:
+    plugin: rust
```
Comment (acatangiu) [inline snap/snapcraft.yaml]: This will use latest rust, you should specify rust version 1.32 using `rust-revision`.

Comment (petreeftime) [reply snap/snapcraft.yaml]: It seems that at some point rustup.sh removed the --prefix option, favoring the environment variables, and this needs to be fixed on the snapcraft side of things: https://github.com/rust-lang/rustup.rs/issues/1703.

**Comment by dianpopa** [comment]
@mikeroyal are you still interested in driving this PR?

**Comment by mikeroyal** [comment]
Hi @dianpopa, I'm getting back to work on it this week. Thank you for the reminder. :)

**Comment by dianpopa** [comment]
I found two other problems that we need to solve:
* The jailer is not installed when doing `snap install`.
* The firecracker binary obtained as part of the built process is built with the gnu target. We want to publish the musl built binary.

**Comment by mikeroyal** [comment]
Hi @dianpopa, I'll look into adding jailer and musl built binary for the snap.

**Comment by dianpopa** [comment]
> Hi @dianpopa, I'll look into adding jailer and musl built binary for the snap.

Unfortunately seems that the 'gnu' target is hardcoded inside the [sources](https://github.com/snapcore/snapcraft/blob/master/snapcraft/plugins/rust.py).
I will open an issue to get their opinion on changing this.

**Comment by dianpopa** [comment]
> > Hi @dianpopa, I'll look into adding jailer and musl built binary for the snap.
> 
> Unfortunately seems that the 'gnu' target is hardcoded inside the [sources](https://github.com/snapcore/snapcraft/blob/master/snapcraft/plugins/rust.py).
> I will open an issue to get their opinion on changing this.

Seems that there was an [effort](https://github.com/snapcore/snapcraft/pull/1382) sometime to add support for musl enabled builds for the Rust plugin but exceeded performance limitations for them.

**Comment by dianpopa** [comment]
Received [response](https://forum.snapcraft.io/t/supoort-for-various-build-targets-for-the-rust-plugin/12474/2) from Snap Team.

**Comment by dianpopa** [comment]
Closing this for now. Will get back to this when I have the time.

**Pull request closed without merging**